### ISSUE TITLE
DualVLA: Building a Generalizable Embodied Agent via Partial Decoupling of Reasoning and Action

### ISSUE THREAD

**Issue by ljwoo94** [ljwoo94/PaperLogs#12]
## What 
- ref : https://arxiv.org/abs/2511.22134
- https://costaliya.github.io/DualVLA/

## Summary
- 이 논문은 로봇의 추론(Reasoning) 능력을 강화할 때 발생하는 행동 성능 저하(Action Degeneration) 문제를 해결하기 위해, 데이터 레벨의 **이중 레이어 프루닝(Pruning)** 과 모델 레벨의 **이중 교사 적응형 증류(Dual-Teacher Distillation)** 를 통해 추론과 행동 학습을 부분적으로 분리한 DualVLA 프레임워크를 제안합니다.

<img width="1029" height="566" alt="Image" src="https://github.com/user-attachments/assets/213918a5-24c5-46ca-ac9b-d8c6c0bda14f" />

**Comment by ljwoo94** [comment]
## Prerequisite 
#### Specialist VLA vs. Reasoning VLA:
- Specialist VLA: 
	- 로봇의 조작 데이터로만 미세 조정되어 행동 능력은 뛰어나지만, 일반적인 상황 판단이나 복잡한 추론 능력은 부족합니다. 
- Reasoning VLA: 
	- 행동 단계마다 "왜 이 행동을 하는지"에 대한 생각(CoT, Chain-of-Thought)을 함께 학습시킨 모델입니다. 일반화 능력은 좋으나 실제 행동 정확도가 떨어지는 경향이 있습니다. 

#### 행동 퇴화 (Action Degeneration): 
- 이 논문에서 정의한 핵심 문제로, 모델에 추론 능력을 주입하기 위해 데이터를 섞어 학습시켰더니, 오히려 기존의 정교한 로봇 조작 능력이 손상되는 현상을 말합니다.

### 기존 연구의 한계: 
- 왜 행동 성능이 떨어지는가?
- **데이터의 중복성**: 
	- 로봇 데이터에 포함된 추론 문장(예: "컵으로 손을 뻗는다")은 매우 단순하고 반복적입니다. 
	- 학습 시 모델이 이 '쉬운 언어 패턴'에만 집중(Overfitting)하게 되어, 정작 복잡한 '실제 팔 움직임(Visuomotor)' 학습을 방해합니다. 
- **감독 신호의 충돌**: 
	- 추론 학습용 신호와 정교한 조작 학습용 신호가 모델 내부에서 서로 충돌하며, 세밀한 제어 성능을 갉아먹습니다

**Comment by ljwoo94** [comment]
## Methodology
- DualVLA는 '추론'과 '행동'을 적절히 **분리(Decoupling)**하여 학습하는 두 가지 핵심 전략을 사용함

### Dual-Layers Data Pruning
- 모든 장면에 추론 데이터를 넣는 대신, 정말 **추론이 필요한 순간**의 데이터만 남김.
- Scene Level pruning
	- 영상의 장면 변화가 큰 '**이벤트 경계**'를 감지하여, 상황이 바뀔 때만 추론을 유지함.
- Action Level Pruning
	- 로봇 팔의 속도가 급변하거나 그리퍼(집게) **상태가 바뀔 때**(예: 물건을 집는 순간)를 핵심 프레임으로 간주함.
	- End effector의 속도가 급격히 변하거나, 그리퍼가 열리고 닫히거나
- 결과: 반복적이고 정보량이 적은 추론 데이터를 제거하여 모델이 실제 행동 학습에 더 집중하게 만듬

### Dual-Teacher Adaptive Distillation
- 행동 교사 (Action Teacher): 
	- 조작 능력이 뛰어난 'Specialist VLA'로부터 정교한 움직임을 배움.
- 추론 교사 (Reasoning Teacher): 
	- 풍부한 상식과 논리력을 가진 기본 VLM 혹은 초기화 모델을 교사로 설정.
- Adaptive Strategy: 
	- 로봇 관련 데이터는 행동 교사에게, 일반 상식 데이터는 추론 교사에게 배우도록 Loss Function을 설계.
	- 데이터의 종류에 따라 어떤 교사의 지도를 받을지 자동으로 결정하여 학습 충돌을 방지.

### VLA Score
- 단순히 "성공/실패"만 따지던 기존 방식에서 벗어나, GPT-4o를 평가자로 활용하여 추론 점수, 행동 점수, 의도 점수, 정렬 점수의 4가지 차원을 종합 평가.
	- Distillation의 경우를 의미하는듯.

**Comment by ljwoo94** [comment]
## Experiment & Result
### 실험 환경
- 시뮬레이션: SimplerEnv (Google Robot, WidowX Robot 환경) 
- 실제 환경: Galaxea R1-lite 로봇 (물체 이동, 물체 전달 작업) 
- 비교 대상: OpenVLA, RT-1/2-X, ECoT, Emma-X 등

<img width="1010" height="533" alt="Image" src="https://github.com/user-attachments/assets/403ce617-095a-4c8a-9cbd-d861f8cb0936" />


환경 | 모델 | 평균 성공률/점수 | 비고
-- | -- | -- | --
SimplerEnv | DualVLA | 61.0% | 기존 최고 모델(ThinkACT, 57.1%) 대비 향상
Real-World | DualVLA | 60.0% | 베이스라인(45.0%) 대비 대폭 향상
VLA Score | DualVLA | 42.9 | 추론과 행동의 균형에서 최고점 달성

**Comment by ljwoo94** [comment]
## Limitation
### 복잡성
- 두 명의 교사 모델(Pre-trained)을 사용해야 하므로, 학습 파이프라인의 구조가 다소 복잡해짐.

### training Overhead
- 학습 과정에서 교사 모델들을 통과시키는 추가적인 Forward pass 연산량이 발생

### Historical Context 부족
- 현재 모델은 과거의 행동 맥락을 충분히 반영하지 못해, 때때로 일관성 없는 의사결정을 내림.

**Comment by ljwoo94** [comment]
#### 학습의 방법론적인 부분인데, 이는 G3에 문의 및 확인 부탁해보자.
